paddle/fluid/operators/ops_extra_info.h

```diff
@@ -14,11 +14,139 @@
 
 #pragma once
 
+#include <string>
+#include <unordered_map>
+#include <vector>
+
 #include "paddle/fluid/framework/attribute.h"
 
 namespace paddle {
 namespace operators {
 
+// This file is to be compatible with the bad design and
+// implementation of fluid in the past
+
+// Many operators in fluid have extra attributes, which are generally added
+// to implement some specific kernel selection and to meet the specialization
+// needs of a specific operation library like mkldnn or cudnn
+enum class ExtraAttrProperty : uint8_t {
+  // The attributes that are no longer used by any scene
+  DEPRECATED = 0,
+  // The attributes used for framework execution scheduling,
+  // such as `use_mkldnn`, `use_cudnn`, no need to save
+  SCHEDULE,
+  // The attributes for ONEDNN only, can be saved in OneDNNContext
+  ONEDNN,
+  // The attributes for ONEDNN only, can be saved in GPUContext
+  GPUDNN,
+  // Add necessary properties as needed
+};
+
+class ExtraAttrPropertySet final {
+ public:
+  constexpr ExtraAttrPropertySet() : bitset_(0) {}
+  constexpr ExtraAttrPropertySet(ExtraAttrProperty e)  // NOLINT
+      : bitset_(e == ExtraAttrProperty::DEPRECATED
+                    ? 0
+                    : 1ULL << (static_cast<uint8_t>(e) - 1)) {}
+
+  inline uint64_t bitset() const { return bitset_; }
+
+  bool inline Support(ExtraAttrProperty e) const {
+    // DEPRECATED ExtraAttr always return false
+    return static_cast<bool>(bitset_ & ExtraAttrPropertySet(e).bitset());
+  }
+  bool IsEmpty() const { return bitset_ == 0; }
+
+  ExtraAttrPropertySet operator|(const ExtraAttrPropertySet& other) const {
+    return ExtraAttrPropertySet(bitset_ | other.bitset());
+  }
+  ExtraAttrPropertySet operator&(const ExtraAttrPropertySet& other) const {
+    return ExtraAttrPropertySet(bitset_ & other.bitset());
+  }
+  ExtraAttrPropertySet operator-(const ExtraAttrPropertySet& other) const {
+    return ExtraAttrPropertySet(bitset_ & ~other.bitset());
+  }
+  ExtraAttrPropertySet operator^(const ExtraAttrPropertySet& other) const {
+    return ExtraAttrPropertySet(bitset_ ^ other.bitset());
+  }
+
+  bool operator==(const ExtraAttrPropertySet& other) const {
+    return bitset_ == other.bitset();
+  }
+
+ private:
+  constexpr ExtraAttrPropertySet(uint64_t bitset) : bitset_(bitset) {}
+  uint64_t bitset_;
+};
+
+const std::unordered_map<std::string, ExtraAttrPropertySet>
+    extra_attr_properties = {
+        // DEPRECATED attributes
+        {"use_quantizer", ExtraAttrProperty::DEPRECATED},
+        // SCHEDULE attributes
+        {"use_cudnn", ExtraAttrProperty::SCHEDULE},
+        {"use_mkldnn", ExtraAttrProperty::SCHEDULE},
+        // ONEDNN dedicated attributes
+        {"Bias", ExtraAttrProperty::ONEDNN},
```

Comment:
Can new extra oneDNN-specific attributes be added in future?

Reply:
In the process of kernel migration, if there are missing attributes, you can add them temporarily . However, in the long run, it is recommended to add fusion op and kernel, and replace the standard op with fusion op in the pass processing stage to support additional functions, so as to avoid the increasingly complex implementation of the basic Op kernel.